tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Reporter.cs

```diff
@@ -62,68 +65,93 @@ private static void LogAddressIfDebugEnabled(IDictionary<string, object> args)
         }
     }
 
-    public void Report(string triggerData, ulong aggregatedTotalRuntime, ulong aggregatedTotalRuntimeWithBindings, bool blocked, int? status = null)
+    internal static void ReportWafInitInfoOnce(Security security, Span span)
     {
-        _localRootSpan.SetTag(Tags.AppSecEvent, "true");
-        if (blocked)
+        if (security.WafInitResult is { Reported: false })
```

Comment:
As this is fired in `OnBeginRequest`, could it be a race confition?

Reply:
Where would the race condition happen? `ReportWafInitInfoOnce` hasn't changed, it's just moved lines (the diff view here is a bit confusing), and `TryReport` is not much different from the former `Report` it's just that it's doing the tests if it should report inside, as now it can also decide to report the schema even without a waf's match